src/App.jsx

```diff
@@ -1,45 +1,243 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import './App.css';
 import TodoForm from './features/TodoForm.jsx';
 import TodoList from './features/TodoList/TodoList.jsx';
 
+const url = `https://api.airtable.com/v0/${import.meta.env.VITE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+const token = `Bearer ${import.meta.env.VITE_PAT}`;
 function App() {
   const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isSaving, setIsSaving] = useState(false);
 
-  function addTodo(title) {
-    const newTodo = { id: Date.now(), title: title, isCompleted: false };
-    setTodoList([...todoList, newTodo]);
-  }
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoading(true);
 
-  function completeTodo(id) {
-    const updatedTodos = todoList.map((todo) => {
-      if (todo.id === id) {
-        return { ...todo, isCompleted: true };
+      const options = {
+        method: 'GET',
+        headers: {
+          Authorization: token,
+        },
+      };
+      try {
+        const resp = await fetch(url, options);
+        if (!resp.ok) {
+          throw new Error('Failed to load todos');
```

Comment:
Be sure that your database doesn't return a message that can be accessed via `resp.message` or `resp.error` otherwise this this is fine.